src/parse.zig

```diff
@@ -571,12 +586,27 @@ const Parser = struct {
                             break :parse;
                         },
                         .NewLine => return error.UnexpectedToken,
-                        else => {},
+                        else => {
+                            var i: usize = tok.start;
+                            while (i < tok.end) : (i += 1) {
+                                var c = self.tree.source[i];
+                                switch (c) {
+                                    '\'' => {
```

Comment:
Hmm, I'd love to keep token processing entirely as part of the Tokenizer, rather than it spilling into the parser as well. I think your alternative idea of tokenizing the escapes is a better approach and I think we should try and implement it straight away rather than relying on a half-measure. Do you think you could look into that?

Reply:
Yes... the one thing that bothers me about that so far is that whether something is an escape sequence or not is dependent on whether it is inside of a single quoted string, double quoted string, or neither. I am already keeping track of that state but I think what I'm not liking is that this reference parser doesn't even want to break the token on spaces and commas like you are doing right now... but we could http://ben-kiki.org/ypaste/data/3072/index.html

I'm not sure if that should change or not. But they do keep the escape sequence as an explicit part of the syntax tree, so mirroring that would be a good thing.

Reply:
If the token includes the spaces, then the parser needs to reach into the token to understand it. Same as for escape sequences. So I guess we could continue to break on spaces and commas and just additionally break on escape sequences...

That would make the parser the simplest I suppose.

Reply:
What if we settled for middle-ground and instead of matching on raw chars in the parser, the tokenizer converts escapes into tokens? Then, we could have a similar logic to what you've already introduced in the parser but match on tokens only. I'm thinking of working with a token stream similar to:

```
.SingleQuote,
.Literal
.EscapeSeq
.Space
.Literal
.SingleQuote
```

I think, if I read your comment right, this would be close to what you want, however, it will require more changes in the tokenizer, but should make the parser cleaner. Thoughts?

Reply:
> If the token includes the spaces, then the parser needs to reach into the token to understand it. Same as for escape sequences. So I guess we could continue to break on spaces and commas and just additionally break on escape sequences...
> 
> That would make the parser the simplest I suppose.

I just saw your latest reply, and yep, I agree it's worth a try, if you feel like taking it on!